boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,947 @@
+}
+
+void Io_Efuse_EnableAux2(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableAux2(void)
+{
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_GetAux1Aux2Status(enum Efuse_Status *status)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_STATR_ADDR, (uint16_t *)status);
+}
+
+ExitCode Io_Efuse_GetAux1Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_FAULTR_0_ADDR, (uint16_t *)fault);
+}
+
+ExitCode Io_Efuse_GetAux2Faults(enum Efuse_Fault *fault)
+{
+    return Io_Efuse_Aux1Aux2ReadRegister(SO_FAULTR_1_ADDR, (uint16_t *)fault);
+}
+
+bool Io_Efuse_IsAux1Aux2InFaultMode(void)
+{
+    return HAL_GPIO_ReadPin(FSB_AUX1_AUX2_GPIO_Port, FSB_AUX1_AUX2_Pin);
+}
+
+bool Io_Efuse_IsAux1Aux2InFailSafeMode(void)
+{
+    return HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin);
+}
+
+void Io_Efuse_DelatchAux1Aux2Faults(void)
+{
+    // Delatch the latchable faults by alternating the inputs high-low-high
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX1_GPIO_Port, PIN_AUX1_Pin, GPIO_PIN_SET);
+
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(PIN_AUX2_GPIO_Port, PIN_AUX2_Pin, GPIO_PIN_SET);
+}
+
+float Io_Efuse_GetAux1Current(void)
+{
+    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
+            AUX1_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
+    {
+        return NAN;
+    }
+
+    if (HAL_GPIO_ReadPin(
+            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return Io_CurrentSense_GetAux1Current();
+    }
+
+    return NAN;
+}
+
+float Io_Efuse_GetAux2Current(void)
+{
+    if (Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(
+            AUX2_CURRENT_SENSE_CHANNEL) != EXIT_CODE_OK)
+    {
+        return NAN;
+    }
+
+    if (HAL_GPIO_ReadPin(
+            CUR_SYNC_AUX1_AUX2_GPIO_Port, CUR_SYNC_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return Io_CurrentSense_GetAux2Current();
+    }
+
+    return NAN;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection)
+{
+    uint16_t reg_val = 0x0000;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, &reg_val));
+
+    CLEAR_BIT(reg_val, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(reg_val, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, reg_val));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_Aux1Aux2ConfigureEfuse(void)
+{
+    RETURN_CODE_IF_EXIT_NOT_OK(Io_Efuse_Aux1Aux2ExitFailSafeMode());
+
+    // Global config register
```

Comment:
What are we doing to the config registers here?

Reply:
Configuring it with the options selected here:
https://github.com/UBCFormulaElectric/Consolidated-Firmware/blob/e1fbea14111c9b660da0b9d6ae460464b01fe612/boards/PDM/Src/Io/Io_Efuse.c#L442

Reply:
Made the comments more clear